Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2339,6 +2339,17 @@ else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDev
             e.printStackTrace();
         }
     }
+
+    private boolean needsSession(ServerRequest request) {
```

Comment:
Can we walk through this tomorrow? I don't think this new function ends up with quite the same results as part of the if statement.

Reply:
Updated Logic